### PR TITLE
Freeze requirements

### PR DESCRIPTION
**Pull request by ttung**
Add a makefile rule (ported from the one I wrote for the HCA DSS) to repin requirements.

### CODE REVIEW

**Comment by ambrosejcarr** [comment]
Can you explain what this does?

**Comment by ttung** [comment]
general usage:

1. edits go to REQUIREMENTS.txt.in and REQUIREMENTS-DEV.txt.in
1. `make REQUIREMENTS.txt` or `make REQUIREMENTS-DEV.txt` will build each.  `make refresh_all_requirements` will rebuild both for the lazy ppl.

**Comment by ttung** [comment]
Alternatively, we could just pin a few core requirements (numpy, scikit-image).  Thoughts?

**Comment by ttung** [comment]
Ready for review.

**Comment by ttung** [comment]
Added missing files.  Edited autogenerated warning.

File: README.md [hunk 1]
```diff
@@ -8,23 +8,19 @@ See this [document](https://docs.google.com/document/d/1IHIngoMKr-Tnft2xOI3Q-5rL
...
-pip 10.0.1 from /usr/local/lib/python3.6/site-packages/pip (python 3.6)
-```
-
-While not required, you may wish to set up a [virtualenv](https://virtualenv.pypa.io/en/stable/). To do this, execute:
+Starfish names its dependencies and lists explicit versions, due to sensitivity to subtle algorithm changes.  For that reason, it is strongly encouraged that you set up a [virtualenv](https://packaging.python.org/tutorials/installing-packages/#creating-virtual-environments).  Create a work folder and set up the virtual environment like:
 ```
+% mkdir starfish
+% cd starfish
 % python -m venv .venv
+% .venv/bin/activate
```
Comment (ambrosejcarr) [inline README.md]: needs `source .venv/bin/activate` right?

File: Makefile [hunk 2]
```diff
@@ -21,6 +22,24 @@ mypy:
 docs-%:
 	make -C docs $*
 
+refresh_all_requirements:
```
Comment (ambrosejcarr) [inline Makefile]: the individual make commands work fine, this one throws an error for me. 

```bash
$ make refresh_all_requirements 
[ ! -e .requirements-env ] || exit 1
virtualenv .REQUIREMENTS.txt.in-env
/bin/bash: virtualenv: command not found
make[1]: *** [REQUIREMENTS.txt] Error 127
make: *** [refresh_all_requirements] Error 2
(.venv) 
``` 

I used a pre-existing virtualenv, although it was set up according to your commands below. Could that contribute? 

I'm inside the virtualenv when I call this, as well.

File: README.md (hunk 1 above)
Comment (ttung) [reply README.md]: right.

File: Makefile (hunk 2 above)
Comment (ttung) [reply Makefile]: Let me try an alternative.

**Comment by ambrosejcarr** [comment]
> Alternatively, we could just pin a few core requirements (numpy, scikit-image). Thoughts?

I think this would cause all the problems with pinning with none of the security it should provide. I think it's all or none. :-)

**Comment by joshmoore** [comment]
Down the road, https://github.com/pypa/pipfile may be a more official solution.

**Pull request merged**